R/pool.R

```diff
@@ -19,24 +18,33 @@ NULL
 #' @export
 Pool <- R6Class("Pool",
   public = list(
+
     valid = NULL,
     counters = NULL,
     minSize = NULL,
     maxSize = NULL,
     idleTimeout = NULL,
+    validateTimeout = NULL,
```

Comment:
I just realized how misleading `validateTimeout` is as a name--it sounds like if the validation operation takes longer than x, we'll give up. Perhaps `validationInterval`, which is apparently what Tomcat's connection pool calls it: https://arviarya.wordpress.com/2013/02/15/apache-dbcp-connection-pool-revisited/

BTW, I also stumbled onto this, which hopefully will make you feel good about us taking the time to implement the validation throttling:

> But one thing I overlooked was, the default value for the testOnBorrow is true. So the moment you set the validationQuery to be any non-null value, every connection you borrow from the pool is going to be validated before it is returned to you. Effectively, if my application was executing 10 DB queries for each customer request, underneath there are 10 more validation requests being executed.
>
> I realized this only when I ran some load tests. The performance was so poor that I couldn't belive what is wrong with my code. Especially even under moderate stress, the system was giving believable results.
>
> One middle ground between these two extremes (testOnBorrow and testWhileIdle) is to selectively validate connections once in every N borrows. For e.g. you should be able to ask DBCP to validate a connection once in every 10 borrows. ... **But this facility is NOT available in Apache DBCP now.** And I am not aware of any other connection pool that is providing this facility either.

http://royontechnology.blogspot.com/2009/02/gotchas-with-dbcp.html